src/main/java/uk/ac/ebi/ena/readtools/loader/fastq/DataSpot.java

```diff
@@ -85,20 +84,19 @@
     {
         expected_base_length = -1;
         expected_qual_length = -1;
-        stream_key = null;
+        readIndex = null;
     }
     
     
     // default - no constraints
     public
     DataSpot( QualityNormalizer normalizer, 
-              String            default_stream_attr,
-              DataSpotParams    params )

```

Comment:
This class is designed to work with externally supplied DataSpotParams object. During processing this class reconfigures params which then get used by future instances of this class. Creating a new default params object on each instance is affecting the behavior of this class. You can know more by looking at the failing test:

uk.ac.ebi.ena.readtools.loader.fastq.LoaderTest.testFailed()